test/test-define-x-expander-use-site-scope.rkt

```diff
@@ -0,0 +1,73 @@
+#lang racket
+
+(require syntax/parse
+         syntax/parse/experimental/eh
+         generic-syntax-expanders
+         syntax/stx
+         (for-syntax syntax/parse
+                     racket/syntax)
+         rackunit)
+
+(define-expander-type eh-mixin)
+
+(begin-for-syntax
+  (define eh-post-accumulate (make-parameter #f)))
+
+(define-for-syntax (strip-use-site stx)
+  (define bd
+    (syntax-local-identifier-as-binding (syntax-local-introduce #'here)))
+  (define delta
+    (make-syntax-delta-introducer (syntax-local-introduce #'here) bd))
+  (delta stx 'remove))
+
+(define-syntax define-eh-alternative-mixin
+  (syntax-parser
+    [(_ name ((~literal pattern) pat) ... (~optional (~seq #:post post)))
+     (let ()
+       #`(define-eh-mixin-expander name
+           (λ (_)
+             #,@(if (attribute post)
+                    #`(((eh-post-accumulate) (quote-syntax post)))
+                    #'())
+             (quote-syntax (~or . #,(strip-use-site #'(pat ...)))))))]))
+
+(define-syntax ~no-order
+  (pattern-expander
+   (λ (stx)
+     (syntax-case stx ()
+       [(self pat ...)
+        (let ()
+          (define acc '())
+          (define (add-to-acc p)
+            (set! acc (cons p acc)))
+          (define alts
+            (parameterize ([eh-post-accumulate add-to-acc])
+              (expand-all-eh-mixin-expanders
+               #'(pat ...))))
+          #`(~and (~seq (~or . #,alts) (... ...))
+                  #,@acc))]))))
+
+;; Test:
+
+(define-eh-alternative-mixin aa
+  (pattern (~optional (~and some-pat #:some))))
+
+(define-eh-alternative-mixin bb
+  (pattern (~optional (~and other-pat #:other)))
+  ;; Without the fix in PR #8, the following line gives the error
+  ;;   attribute: not bound as a pattern variable in: some-pat
+  #:post (~fail #:when (and (attribute some-pat)
```

Comment:
So to make sure I understand the bug: this fails because the definition of `aa` comes with an extra scope separate from the "expected" macro use-site scope, causing binding issues when trying to retrieve the stored identifiers in the eh-post-accumulate parameter, and that scope doesn't serve a useful purpose because it's just doing the exact same thing a macro-introduction scope would do. Additionally, it's very hard to remove that scope inside the macro definition, so it would be better to just not add it. Is that about right?